### PR TITLE
masher should clean up old mashes at the end of every run

### PR DESCRIPTION
**Pull request by ankit01ojha**
fixes #https://github.com/fedora-infra/bodhi/issues/1304

### CODE REVIEW

**Comment by ankit01ojha** [comment]
@bowlofeggs Can you please review this.

File: bodhi/server/consumers/masher.py [hunk 1]
```diff
@@ -979,3 +981,41 @@ def run(self):
...
+    for dirs in pattern_matched_dirs.values():
+        if len(dirs) > NUM_TO_KEEP:
+            dirs_to_delete.extend(sorted(dirs, reverse=True)[NUM_TO_KEEP:])
+
+    if dirs_to_delete:
+        print 'Deleting the following directories:'
+        for d in dirs_to_delete:
+            d = os.path.join(mash_dir, d)
+            shutil.rmtree(d)
+            print d
```
Comment (bowlofeggs) [inline bodhi/server/consumers/masher.py]: This code is duplicated with the script. We don't want to copy code as that violates the DRY Principle:

https://en.wikipedia.org/wiki/Don't_repeat_yourself

Instead, we want to import the existing code that does this and use it from the masher.

File: bodhi/tests/server/consumers/test_masher.py [hunk 2]
```diff
@@ -1231,3 +1235,69 @@ def test_comps_unsafe_http_url(self, mock_cmd):
...
+            actual_files = [f for f in os.listdir(mash_dir)
+                            if os.path.isfile(os.path.join(mash_dir, f))]
+            self.assertEqual(actual_files, ['COOL_FILE.txt'])
+            # Make sure the printed output is correct
+            expected_output = set(dirs) - expected_dirs
+            expected_output = {os.path.join(mash_dir, d) for d in expected_output}
+            expected_output = expected_output | {'Deleting the following directories:', ''}
+            self.assertEqual(set(result.output.split('\n')), expected_output)
+        finally:
+            shutil.rmtree(mash_dir)
```
Comment (bowlofeggs) [inline bodhi/tests/server/consumers/test_masher.py]: This test is copied from the existing test. We don't need to retest that this code works, but we do need to assert that the masher calls it when done.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
I recommend adjusting the commit message to say "mashes" instead of "mashers". Also, the commit message has two #'s in front of the issue number instead of one.

**Comment by ankit01ojha** [comment]
@bowlofeggs I don't know why but the clean_up function is not getting called, I am getting the test failed `AssertionError: 0 != 1`.

**Comment by bowlofeggs** [comment]
@ankit01ojha it's not getting called because nothing is explicitly calling it. Python doesn't run functions when they are defined; it runs them when they are called, i.e., you put ```()```'s after its name.

File: bodhi/server/consumers/masher.py [hunk 3]
```diff
@@ -979,3 +980,7 @@ def run(self):
         else:
             self.success = True
         return out, err, returncode
+
+
+def clean_up():
+    pass
```
Comment (bowlofeggs) [inline bodhi/server/consumers/masher.py]: I dont' think we need a new function here. We just need to call ```clean_old_mashes()``` when the mash is done and successful.

**Comment by bowlofeggs** [comment]
This will also need release notes added at https://github.com/fedora-infra/bodhi/blob/develop/docs/release_notes.rst

**Comment by ankit01ojha** [comment]
done!

File: bodhi/server/consumers/masher.py [hunk 4]
```diff
@@ -974,6 +976,9 @@ def _get_master_repomd_url(self, arch):
 
         return master_repomd % (self.release.version, arch)
 
+    def clean_up(self):
+        pass
```
Comment (bowlofeggs) [inline bodhi/server/consumers/masher.py:980]: This method doesn't do anything. Why put it here?

File: bodhi/tests/server/consumers/test_masher.py [hunk 5]
```diff
@@ -155,6 +155,10 @@ def test___init___sets_bugtracker(self, set_bugtracker):
 
         set_bugtracker.assert_called_once_with()
 
+    @mock.patch('bodhi.server.scripts.clean_old_mashes.NUM_TO_KEEP', 2)
+    def test_clean_up(self, clean_up):
+        self.assertEquals(len(clean_up.call_args_list), 1)
```
Comment (bowlofeggs) [inline bodhi/tests/server/consumers/test_masher.py:160]: This test doesn't run the code and doesn't assert that the clean up call was made.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
This can

**Pull request closed without merging**